bolt-sidecar/src/builder/payload_builder.rs

```diff
@@ -167,6 +146,11 @@ impl FallbackPayloadBuilder {
         let blob_gas_used =
             transactions.iter().fold(0, |acc, tx| acc + tx.blob_gas_used().unwrap_or_default());
 
+        // We must calculate the next block timestamp manually rather than rely on the
+        // previous execution block, to cover the edge case where any previous slots have
+        // been missed by the proposers immediately before us.
+        let block_timestamp = self.genesis_time + (target_slot * self.slot_time);
```

Comment:
this should be the new timestamp because target_slot is the one we are expected to propose at.